### PR TITLE
OAI OpenAIRE4: changes to related identifiers and alternative identifiers

### LINKED ISSUE #3099: OpenAIRE 4 relatedIdentifier being used for all dc:identifier:* metadata other than dc:identifier:uri
**Describe the bug**
datacite:relatedIdentifier element being used for all dc:identifier:* metadata other than dc:identifier:uri. Not all dc:identifier:* values can be assumed to be relatedIdentifier and may be alternateIdentifier instead.

**To Reproduce**
Access an openaire4 OAI record for an item that has many different dc:identifier:* values. Every one other than dc:identifier:uri will show up as datacite:relatedIdentifier elements.

**Expected behavior**
When it isn't known for sure that the dc:identifier:* value is actually a relatedIdentifier value, perhaps it is better to assume it is an alternateIdentifier by default. This follows the approach originally included with the proposed [PR 2709](https://github.com/DSpace/DSpace/pull/2709) for DSpace 6.x that used alternateIdentifier, except for the case of dc:identifier:issn that was handled separately as a relatedIdentifier. Another common use case is with dc:identifier:doi being an externally-generated DOI for the item (i.e., not generated by DSpace automatically) and being an alternateIdentifier to the DSpace Handle. 

**Related work**
See discussions from:
[PR 2576](https://github.com/DSpace/DSpace/pull/2576) (specifically [this comment](https://github.com/DSpace/DSpace/pull/2576#issuecomment-559110676))
[DS-3708](https://jira.lyrasis.org/browse/DS-3708)

### PR DESCRIPTION
**Pull request by paulo-graca**
## References
* Fixes #3099
* related with the work done in #2576

## Description
This PR changes slightly the way related identifiers are exposed in OpenAIRE4 interface accordingly to what it's stated in #PR2576. 

## Instructions for Reviewers
PR #2576 (specifically [this comment](https://github.com/DSpace/DSpace/pull/2576#issuecomment-559110676)). We defined in what way to consider identifiers in openaire's oai context. This Pull Request tries to address that.

List of changes in this PR:
* The way related identifiers are considered and exposed in OpenAIRE4 context
* The way alternative identifiers are considered and exposed in OpenAIRE4 context

In order to test it you first need to ensure you have OAI enabled:
`oai.enabled = true`

also ensure you have content on your OAI interface, you can index it using:
`[/dspace/bin/]dspace oai import -c`

then access any item on your OAI interface within the openaire context:
https://yourserver.com/server/oai/openaire4?verb=GetRecord&metadataPrefix=oai_openaire&identifier=oai:oaicontext.com:123456789/1

### CODE REVIEW

File: dspace/config/crosswalks/oai/metadataFormats/oai_openaire.xsl
Comment (tdonohue) [inline dspace/config/crosswalks/oai/metadataFormats/oai_openaire.xsl:547]: This isn't an out-of-the-box DSpace field.   We should either note that here, or perhaps comment out this template.  If sites choose to add this field, they can uncomment this template to update the OAI-PMH configuration.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@paulo-graca : Overall, this looks fine.  I haven't tested it yet but just reviewed the code quickly.  However, I do think the `dc.identifier.doi` logic should either be commented out or removed.  That's not a field that exists in DSpace by default.  If sites choose to add that field, then the expectation is that they should customize their OAI-PMH crosswalks as they see fit to manage that field.  So, having an example template is fine...but I think it should be disabled by default, as it will never be triggered by default.

@uofmsean, would you be willing to give this PR a test for us to see if it resolves the main issues you reported?

File: dspace/config/crosswalks/oai/metadataFormats/oai_openaire.xsl
Comment (uofmsean) [reply dspace/config/crosswalks/oai/metadataFormats/oai_openaire.xsl:547]: There are also cases where it can be treated as an alternateIdentifier if manually assigned from outside DSpace. Perhaps it is best to leave it up to the repository managers.

File: dspace/config/crosswalks/oai/metadataFormats/oai_openaire.xsl
Comment (uofmsean) [inline dspace/config/crosswalks/oai/metadataFormats/oai_openaire.xsl:485]: Although not a part of this patch's changes, there's an issue with this for-each. It iterates over the dc:identifier:* elements, but this results in only the first one of each being processing based on how the relatedIdentifierTemplate is called. Modifications are needed (somewhere) so that iteration is over all
`./doc:element/doc:element/doc:field[@name='value']/text()`
values. This would handle multiple dc:identifier:* values including language variants and multiple possible values for each one.

File: dspace/config/crosswalks/oai/metadataFormats/oai_openaire.xsl
Comment (paulo-graca) [reply dspace/config/crosswalks/oai/metadataFormats/oai_openaire.xsl:547]: @tdonohue there is also other transformations in place supporting `dc:identifier.doi` like `uketd_dc`:
https://github.com/DSpace/DSpace/blob/main/dspace/config/crosswalks/oai/metadataFormats/uketd_dc.xsl#L221-L223

I don't see any major implications for leaving it. 
But I think to remove it from here, for consistency purposes, all other existing references to `dc:identifier.doi` should also be removed. What do you think?

Comment (tdonohue) [reply dspace/config/crosswalks/oai/metadataFormats/oai_openaire.xsl:547]: @paulo-graca : Good catch, I was not aware this had been done previously as well.  I'm OK then with letting this in, if we feel there are users who would make use of "dc.identifier.doi".   It just seems a bit odd to add in logic not relating to an out-of-the-box DSpace metadata field...though maybe it's a field we should be considering.

In any case, I'm OK with leaving this in, as you've proven we've done so previously.

**Comment by paulo-graca** [comment]
@uofmsean thank you so much for your thoroughly testing. I've made some recent changes to address the issue you reported. Can you re-test it please?

**Review by tdonohue — APPROVED** [review]
👍 This looks good to me now.  Thanks @paulo-graca !   @uofmsean , did you want to give this one last test on your end before we merge it?

**Comment by tdonohue** [comment]
Merging this as-is, as it was flagged for `1 Approval`.

**Pull request merged**
